help/Readme.md

```diff
@@ -0,0 +1,114 @@
+```
+
+Download Virtual-Environments repository.
+```
+git clone https://github.com/actions/virtual-environments.git
+```
+
+Import [GenerateResourcesAndImage](https://github.com/actions/virtual-environments/blob/main/helpers/GenerateResourcesAndImage.ps1) script from `/helpers` folder, and run `GenerateResourcesAndImage` function in Powershell.
+
+```
+Import-Module C:\virtual-environments\helpers\GenerateResourcesAndImage.ps1
+
+GenerateResourcesAndImage -SubscriptionId {YourSubscriptionId} -ResourceGroupName "myTestResourceGroup" -ImageGenerationRepositoryRoot "C:\virtual-environments" -ImageType Ubuntu1604 -AzureLocation "East US"
+```
+Where:
+- `SubscriptionId` - The Azure subscription Id where resources will be created.
+- `ResourceGroupName` - The Azure resource group name where the Azure resources will be created.
+- `ImageGenerationRepositoryRoot` - The root path of the image generation repository source.
+- `ImageType` - The type of the image being generated. Valid options are: "Windows2016", "Windows2019", "Ubuntu1604", "Ubuntu1804", "Ubuntu2004".
+- `AzureLocation` - The location of the resources being created in Azure. For example "East US".
+
+Function will automatically create all required Azure resources and kick off a packer image generation for the selected image type.
+
+*Please, check synopsis of `GenerateResourcesAndImage` for details about non mandatory parameters.*
+
+#### Generated VM Deployment
+After successful image generation, you can deploy Virtual Machine based on generated VHD using [CreateAzureVMFromPackerTemplate](https://github.com/actions/virtual-environments/blob/main/helpers/CreateAzureVMFromPackerTemplate.ps1) script.
+```
+Import-Module C:\virtual-environments\helpers\CreateAzureVMFromPackerTemplate.ps1
+
+CreateAzureVMFromPackerTemplate -SubscriptionId {YourSubscriptionId}  -ResourceGroupName {ResourceGroupName} -TemplateFile "C:\BuildVmImages\temporaryTemplate.json" -VirtualMachineName "testvm1" -AdminUsername "shady1" -AdminPassword "SomeSecurePassword1" -AzureLocation "eastus"
+```
+Where:
+- `SubscriptionId` - The Azure subscription Id where resources will be created.
+- `ResourceGroupName` - The Azure resource group name where the Azure virtual machine will be created.
+- `TemplatFilePath` - The path to the json template generated by packer during image generation locally.
+- `VirtualMachineName` - The name of the virtual machine to be generated.
+- `AdminUserName` - The administrator username for the virtual machine to be created.
+- `AdminPassword` - The administrator password for the virtual machine to be created.
+- `AzureLocation` - The location where the Azure virtual machine will be provisioned. Example: "eastus"
+
+Fuction will create an Azure VM from a template and generate network resources in Azure to make the VM accessible.
+
+### Additional
+#### Builder variables
+Each packer template includes `variables` group that contains builder variables that are used in image generation.
```

Comment:
Actually there are 2 sets:
variables —  is an object of one or more key/value strings that defines user variables contained in the template
builders — is an array of one or more objects that defines the builders that will be used to create machine images for this template, and configures each of those builders.